bartop-ui/src/components/Auth/Auth.js

```diff
@@ -0,0 +1,126 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Switch, Route } from 'react-router-dom';
+
+import LoginForm from './LoginForm/LoginForm';
+import VerifyForm from './VerifyForm/VerifyForm';
+import Callback from './Callback/Callback';
+import Failure from './Failure/Failure';
+import { actions } from '../../ducks/authentication/authentication';
+import NotFound from '../NotFound/NotFound';
+
+export class Auth extends Component {
+  constructor() {
+    super();
+    this.sendCode = this.sendCode.bind(this);
+    this.verifyCode = this.verifyCode.bind(this);
+  }
+
+  async sendCode(email) {
+    return new Promise((resolve, reject) => {
+      this.props.auth.passwordlessStart(
+        {
+          connection: 'email',
+          send: 'code',
+          email
+        },
+        (err, res) => {
+          if (err) {
+            reject(err);
+          } else {
+            resolve(res);
+          }
+        }
+      );
+    });
+  }
+
+  async verifyCode(email, verificationCode) {
+    return new Promise((resolve, reject) => {
+      this.props.auth.passwordlessLogin(
+        {
+          connection: 'email',
+          email,
+          verificationCode
+        },
+        (err, res) => {
+          if (err) {
+            reject(err);
+          } else {
+            resolve(res);
+          }
+        }
+      );
+    });
+  }
+
+  render() {
+    const { match, handleAuthentication } = this.props;
+
+    return (
+      <Switch>
+        <Route
+          exact
+          path={`${match.url}/login`}
+          render={({ history, location }) => {
+            // if the user came back from verify, we will have their email in location state so we can prefill
+            const email =
+              location.state && location.state.email
+                ? location.state.email
+                : undefined;
+            return (
+              <LoginForm
+                history={history}
+                sendCode={this.sendCode}
+                prefillEmail={email}
+              />
+            );
+          }}
+        />
+        <Route
+          exact
+          path={`${match.url}/verify`}
+          render={({ history, location }) =>
+            location.state && location.state.email ? (
+              <VerifyForm
+                history={history}
+                email={location.state.email}
+                verifyCode={this.verifyCode}
+              />
+            ) : (
+              <NotFound />
```

Comment:
This is saying they get a `Page Not Found` if their email is not saved in the state after verification?

Reply:
They get a `404` if they reach the `bartop.io/auth/verify route without having an `email` in the state, which shouldn't ever happen because they need to hit `bartop.io/auth/login` first, which is where the code is sent and the `email` is set. The only way they can reach `verify` w/o the email is if they manually typed in the URL, which in that case "wouldn't be found" since its not valid w/o the email

Reply:
Bingo